NONE - Closed in favor of PR #5910

mrsrinivas opened the pull request:
Closed in favor of [PR #5910](https://github.com/apache/kafka/pull/5910)

mrsrinivas committed:
KAFKA-7418: Add '--help' option to missing Kafka CLI commands (KIP-374)

mrsrinivas committed:
KAFKA-7418: Add '--help' option to missing Kafka CLI commands (KIP-374)

mrsrinivas commented:
retest this please

mrsrinivas committed:
KAFKA-7418: Add '--help' option to missing Kafka CLI commands (KIP-374)

mrsrinivas committed:
KAFKA-7418: Add '--help' option to connect commands

mimaison commented on connect/runtime/src/main/java/org/apache/kafka/connect/cli/ConnectDistributed.java:
```diff
@@ -27,12 +27,7 @@
-import org.apache.kafka.connect.storage.KafkaOffsetBackingStore;
-import org.apache.kafka.connect.storage.KafkaStatusBackingStore;
-import org.apache.kafka.connect.storage.StatusBackingStore;
+import org.apache.kafka.connect.storage.*;
```
Considering we're only using a few classes from this package, should we just keep the individual imports instead of using star here?

mimaison commented on core/src/main/scala/kafka/admin/PreferredReplicaLeaderElectionCommand.scala:
```diff
@@ -1,68 +1,51 @@
+  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  * See the License for the specific language governing permissions and
+  * limitations under the License.
+  */
```
Do we want this? Javadocs are usually not aligned this way. Same in a few other classes.

If we want to touch the license block, we should maybe use regular comments instead of javadoc

mrsrinivas commented:
retest this please

mimaison commented on core/src/main/scala/kafka/admin/TopicCommand.scala:
```diff
@@ -21,18 +21,17 @@ import java.util.Properties
 import kafka.server.ConfigType
-import kafka.utils._
+import kafka.utils.Implicits._
+import kafka.utils.{Whitelist, _}
```
Can we simplify that to `kafka.utils._`?

mimaison commented on core/src/main/scala/kafka/tools/MirrorMaker.scala:
```diff
@@ -29,37 +29,37 @@ import joptsimple.OptionParser
 import kafka.metrics.KafkaMetricsGroup
 import kafka.utils.{CommandLineUtils, CoreUtils, Logging, Whitelist}
-import org.apache.kafka.clients.consumer.{CommitFailedException, Consumer, ConsumerConfig, ConsumerRebalanceListener, ConsumerRecord, KafkaConsumer, OffsetAndMetadata}
+import org.apache.kafka.clients.consumer._
```
Is this needed? It does not look like we're using any new classes in this package

mimaison commented on core/src/main/scala/kafka/utils/CommandLineUtils.scala:
```diff
@@ -1,65 +1,88 @@
+    * @return true on matching the help check condition
+    */
+  def checkHelpArg(commandOpts: CommandDefaultOptions): Boolean = {
+    return (commandOpts.args.length == 0 || commandOpts.options.has(commandOpts.helpOpt))
```
`return` and the outer braces are not needed here. Also should we find a better name for this method, `checkHelpArg` is a bit confusing with the `Boolean` return type.

mimaison reviewed (changes requested):
Not sure about all the javadoc alignment changes. These look unecessary. Apart from that it looks good overal, I just left a few comments for minor issues

mrsrinivas committed:
KAFKA-7418: addressed review comments

mrsrinivas commented:
Thank you @mimaison, for the review. Most of the import and documentation changes happen due IDEA auto format while commit the code. 

Now I addressed the all points.

mrsrinivas commented:
retest this please

mrsrinivas committed:
Merge pull request #1 from apache/trunk

merge

mrsrinivas committed:
Revert "merge"

mrsrinivas committed:
Merge pull request #2 from mrsrinivas/revert-1-trunk

Revert "merge"

mrsrinivas commented:
@mimaison, I messed up this branch accidentally when I was trying to incorporate the warning fixed from trunck. I raised new [PR #5910 here](https://github.com/apache/kafka/pull/5910) and addressed your comments.

Thanks again for the feedback.
